### ISSUE TITLE
Panic in plane splitting code in CSS parallax demo

### ISSUE THREAD

**Issue by mrobinson** [servo/webrender#1444]
When running http://keithclark.co.uk/articles/pure-css-parallax-websites/demo3/ in Servo I see this panic:

```
no entry found for key (thread RenderBackend, at /checkout/src/libcore/option.rs:823)
stack backtrace:
   0:     0x556f56107484 - backtrace::backtrace::libunwind::trace
                        at /home/martin/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.2/src/backtrace/libunwind.rs:53
                         - backtrace::backtrace::trace<closure>
                        at /home/martin/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.2/src/backtrace/mod.rs:42
   1:     0x556f560ff92f - backtrace::capture::{{impl}}::new
                        at /home/martin/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.2/src/capture.rs:64
   2:     0x556f50704b56 - servo::main::{{closure}}
                        at /home/martin/work/servo/ports/servo/main.rs:129
   3:     0x556f561cc5ea - std::panicking::rust_panic_with_hook
                        at /checkout/src/libstd/panicking.rs:550
   4:     0x556f561cc484 - std::panicking::begin_panic<collections::string::String>
                        at /checkout/src/libstd/panicking.rs:511
   5:     0x556f561cc3b9 - std::panicking::begin_panic_fmt
                        at /checkout/src/libstd/panicking.rs:495
   6:     0x556f561cc347 - std::panicking::rust_begin_panic
                        at /checkout/src/libstd/panicking.rs:471
   7:     0x556f562038fd - core::panicking::panic_fmt
                        at /checkout/src/libcore/panicking.rs:69
   8:     0x556f5620396d - core::option::expect_failed
                        at /checkout/src/libcore/option.rs:823
   9:     0x556f5468fce4 - core::option::{{impl}}::expect<&webrender::render_task::RenderTask>
                        at /checkout/src/libcore/option.rs:302
  10:     0x556f54632067 - std::collections::hash::map::{{impl}}::index<webrender::tiling::StackingContextIndex,webrender::tiling::StackingContextIndex,webrender::render_task::RenderTask,std::collections::hash::map::RandomState>
                        at /checkout/src/libstd/collections/hash/map.rs:1332
  11:     0x556f5474266f - webrender::frame_builder::{{impl}}::build_render_task
                        at /home/martin/work/mozilla/webrender/webrender/src/frame_builder.rs:1302
  12:     0x556f54743f40 - webrender::frame_builder::{{impl}}::build
                        at /home/martin/work/mozilla/webrender/webrender/src/frame_builder.rs:1406
  13:     0x556f545f3eee - webrender::frame::{{impl}}::build_frame::{{closure}}
                        at /home/martin/work/mozilla/webrender/webrender/src/frame.rs:1178
  14:     0x556f5457bcfc - core::option::{{impl}}::map<&mut webrender::frame_builder::FrameBuilder,webrender::tiling::Frame,closure>
                        at /checkout/src/libcore/option.rs:398
  15:     0x556f545f3c26 - webrender::frame::{{impl}}::build_frame
                        at /home/martin/work/mozilla/webrender/webrender/src/frame.rs:1177
  16:     0x556f545f3972 - webrender::frame::{{impl}}::build
                        at /home/martin/work/mozilla/webrender/webrender/src/frame.rs:1155
  17:     0x556f548eb87f - webrender::render_backend::{{impl}}::render
                        at /home/martin/work/mozilla/webrender/webrender/src/render_backend.rs:514
  18:     0x556f54758937 - webrender::render_backend::{{impl}}::run::{{closure}}
                        at /home/martin/work/mozilla/webrender/webrender/src/render_backend.rs:434
  19:     0x556f5475326a - webrender::profiler::{{impl}}::profile<webrender::internal_types::RendererFrame,closure>
                        at /home/martin/work/mozilla/webrender/webrender/src/profiler.rs:154
  20:     0x556f548ea469 - webrender::render_backend::{{impl}}::run
                        at /home/martin/work/mozilla/webrender/webrender/src/render_backend.rs:433
  21:     0x556f548f0393 - webrender::renderer::{{impl}}::new::{{closure}}
                        at /home/martin/work/mozilla/webrender/webrender/src/renderer.rs:1158
  22:     0x556f547e271a - std::sys_common::backtrace::__rust_begin_short_backtrace<closure,()>
                        at /checkout/src/libstd/sys_common/backtrace.rs:136
  23:     0x556f546a5f93 - std::thread::{{impl}}::spawn::{{closure}}::{{closure}}<closure,()>
                        at /checkout/src/libstd/thread/mod.rs:364
  24:     0x556f5472214a - std::panic::{{impl}}::call_once<(),closure>
                        at /checkout/src/libstd/panic.rs:296
  25:     0x556f546a7bc9 - std::panicking::try::do_call<std::panic::AssertUnwindSafe<closure>,()>
                        at /checkout/src/libstd/panicking.rs:454
  26:     0x556f561d374a - panic_unwind::__rust_maybe_catch_panic
                        at /checkout/src/libpanic_unwind/lib.rs:98
  27:     0x556f546a6f3c - std::panicking::try<(),std::panic::AssertUnwindSafe<closure>>
                        at /checkout/src/libstd/panicking.rs:433
  28:     0x556f546a5cc5 - std::panic::catch_unwind<std::panic::AssertUnwindSafe<closure>,()>
                        at /checkout/src/libstd/panic.rs:361
  29:     0x556f54811a2b - std::thread::{{impl}}::spawn::{{closure}}<closure,()>
                        at /checkout/src/libstd/thread/mod.rs:363
  30:     0x556f5484efb3 - alloc::boxed::{{impl}}::call_box<(),closure>
                        at /checkout/src/liballoc/boxed.rs:648
  31:     0x556f561cb245 - alloc::boxed::{{impl}}::call_once<(),()>
                        at /checkout/src/liballoc/boxed.rs:658
                         - std::sys_common::thread::start_thread
                        at /checkout/src/libstd/sys_common/thread.rs:21
                         - std::sys::imp::thread::{{impl}}::new::thread_start
                        at /checkout/src/libstd/sys/unix/thread.rs:84
  32:     0x7fce4d11a6d9 - start_thread
  33:     0x7fce4cc3dd7e - __clone
  34:                0x0 - <unknown>
```

The code in question is:
```
                    if !preserve_3d_map.is_empty() && parent_isolation != Some(ContextIsolation::Items) {
                        // Flush the accumulated plane splits onto the task tree.
                        // Notice how this is done before splitting in order to avoid duplicate tasks.
                        current_task.children.extend(preserve_3d_map.values().cloned());
                        debug!("\tplane splitting in {:?}", current_task.id);
                        // Z axis is directed at the screen, `sort` is ascending, and we need back-to-front order.
                        for poly in splitter.sort(vec3(0.0, 0.0, 1.0)) {
                            let sc_index = StackingContextIndex(poly.anchor);
>>>>>
                            let task_id = preserve_3d_map[&sc_index].id;
<<<<
                            debug!("\t\tproduce {:?} -> {:?} for {:?}", sc_index, poly, task_id);
                            let pp = &poly.points;
                            let gpu_blocks = [
                                [pp[0].x, pp[0].y, pp[0].z, pp[1].x].into(),
                                [pp[1].y, pp[1].z, pp[2].x, pp[2].y].into(),
                                [pp[2].z, pp[3].x, pp[3].y, pp[3].z].into(),
                            ];
                            let handle = gpu_cache.push_per_frame_blocks(&gpu_blocks);
                            let item = AlphaRenderItem::SplitComposite(sc_index, task_id, handle, next_z);
                            current_task.as_alpha_batch().items.push(item);
                        }
                        splitter.reset();
                        preserve_3d_map.clear();
                        next_z += 1;
                    }
                }
```

**Comment by mrobinson** [comment]
cc @kvark

**Comment by kvark** [comment]
Fixed by  #1448
